### DISCUSSION TITLE
can't able to push the changes to github repo

### DISCUSSION THREAD

**Discussion by Yash26112002** [community/community#51886] [Actions]
### Select Topic Area

Question

### Body

actually i can't able to push changes to github repo as the error is popping out many time

**remote: Permission to "my username"/"my-repo".git denied to myfriend-username
fatal: unable to access 'https://github.com/"my -username"/"my-repo".git': The requested URL returned error: 403**

actually i have logged in from my friend github account on my pc and i have may be added new ssh key to my system i think that my my friend account is linked to my git

i have done removing ssh keys
i have removed credentials
i have done everything with CHATGPT or StackOverflow everything i done but the same error is popping out

plzzz tell me

**Comment by airtower-luna** [comment]
What I can see from the error message:

* The remote URL uses HTTPS, so SSH keys have nothing to do with it.
* You're not asked for credentials, so there must be a token cached *somewhere*.

Where that might be depends on your Git settings and OS.

And don't ask ChatGPT, it just generates stuff that looks similar to human writing, but it doesn't know anything.

**Reply by Yash26112002** [reply]
so how to resolve the issue can you plzz guide me

**Reply by Yash26112002** [reply]
i also run this cmnd **git config --unset credential.helper** but it doesn't work and the same error is popping out
and the config list using the cmnd **git config --list --show-origin** is

PS C:\Users\YASH SANGWAN\Desktop\hackathon\LifeLy> git config --list --show-origin
file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain
file:C:/Program Files/Git/etc/gitconfig filter.lfs.clean=git-lfs clean -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.smudge=git-lfs smudge -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.process=git-lfs filter-process
file:C:/Program Files/Git/etc/gitconfig filter.lfs.required=true
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=openssl
file:C:/Program Files/Git/etc/gitconfig http.sslcainfo=C:/Program Files/Git/mingw64/etc/ssl/certs/ca-bundle.crt
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=true
file:C:/Program Files/Git/etc/gitconfig core.fscache=true
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false
file:C:/Program Files/Git/etc/gitconfig core.fsmonitor=true
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false
file:C:/Program Files/Git/etc/gitconfig credential.helper=manager
file:C:/Program Files/Git/etc/gitconfig credential.https://dev.azure.com.usehttppath=true
file:C:/Program Files/Git/etc/gitconfig init.defaultbranch=master
file:C:/Users/YASH SANGWAN/.gitconfig   url.https://github.com/.insteadof=git://github.com/
file:C:/Users/YASH SANGWAN/.gitconfig   user.name="my name"
file:C:/Users/YASH SANGWAN/.gitconfig   user.email="my email"
file:C:/Users/YASH SANGWAN/.gitconfig   push.default=simple
file:C:/Users/YASH SANGWAN/.gitconfig   credential.helper=wincred
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=false
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.symlinks=false
file:.git/config        core.ignorecase=true
file:.git/config        user.email="my email"
file:.git/config        user.name="my name"
file:.git/config        remote.origin.url=https://github.com/Yash26112002/LifeLy.git
:...skipping...
file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain
file:C:/Program Files/Git/etc/gitconfig filter.lfs.clean=git-lfs clean -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.smudge=git-lfs smudge -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.process=git-lfs filter-process
file:C:/Program Files/Git/etc/gitconfig filter.lfs.required=true
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=openssl
file:C:/Program Files/Git/etc/gitconfig http.sslcainfo=C:/Program Files/Git/mingw64/etc/ssl/certs/ca-bundle.crt
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=true
file:C:/Program Files/Git/etc/gitconfig core.fscache=true
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false
file:C:/Program Files/Git/etc/gitconfig core.fsmonitor=true
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false
file:C:/Program Files/Git/etc/gitconfig credential.helper=manager
file:C:/Program Files/Git/etc/gitconfig credential.https://dev.azure.com.usehttppath=true
file:C:/Program Files/Git/etc/gitconfig init.defaultbranch=master
file:C:/Users/YASH SANGWAN/.gitconfig   url.https://github.com/.insteadof=git://github.com/
file:C:/Users/YASH SANGWAN/.gitconfig   user.name="my name"
file:C:/Users/YASH SANGWAN/.gitconfig   user.email="my email"
file:C:/Users/YASH SANGWAN/.gitconfig   push.default=simple
file:C:/Users/YASH SANGWAN/.gitconfig   credential.helper=wincred
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=false
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.symlinks=false
file:.git/config        core.ignorecase=true
file:.git/config        user.email="my email"
file:.git/config        user.name="my name"
file:.git/config        remote.origin.url=https://github.com/Yash26112002/LifeLy.git
file:.git/config        remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*

**Reply by airtower-luna** [reply]
The `git config --list --show-origin` output shows `credential.helper` is still set in your *user* and *system* config. `git config --unset credential.helper` will only unset it in the *repository* config. If you want to unset the credential helper in the user config, you need to use `--global`. There's another flag for the system config, but I don't know if that might mess with other users on your system. So if you want to disable the credential helper, you should actually set it to empty in your user config:

```
git config --global credential.helper ''
```

But then you'll be asked for your username and password on every push, so it might be worth actually [setting up the `cache` helper](https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage), or figuring out where the current `wincred` stores the credentials and fix them there.

**Reply by Yash26112002** [reply]
i have done it but same error is showing

<img width="921" alt="image" src="https://user-images.githubusercontent.com/77532366/231954836-966ca90d-4437-4753-bc1c-d4605a27e132.png">

**Reply by airtower-luna** [reply]
It looks like setting the `credential.helper` didn't work (evidence: the output shows the current value). Probably an issue with Powershell, try to use Git Bash instead.

**Reply by Yash26112002** [reply]
<img width="682" alt="image" src="https://user-images.githubusercontent.com/77532366/234776429-2576d7c1-7f12-4ae6-b963-94bee74c6e61.png">
same error with git bash
i used ssd keys to setup my friend account in my laptop and there are already ssd keys for my account too
may be this info helps

**Reply by airtower-luna** [reply]
I meant using Git Bash to configure the credential helper. :wink: 

The output of `git push` shows that your repository is configured with an HTTPS URL. If you push over HTTPS Git can't use your SSH keys. If you have the SSH key set up already, all you need to do is change the URL to the SSH URL for your repository using [`git remote set-url`](https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-urlem), like so:

```
git remote set-url origin git@github.com:USER/REPO.git
```

You can find the right URL on the repository page. Of course like the token the SSH key you use needs to be for the right account. :sweat_smile:

**Comment by lalajia** [comment]
Hi, do you find the solution to your problem? I have exactly the same issue!
